NwPluginAPI/Core/Extensions/ItemExtensions.cs

```diff
@@ -0,0 +1,220 @@
+		/// <param name="attachmentCode">The attachments code representing the set of attachments to apply.</param>
+		public static void ApllyAttachments(this Firearm firearm, uint attachmentCode)
+		{
+			if (firearm == null)
+				throw new ArgumentNullException(nameof(Firearm), "Firearm is null");
+
+			firearm.ApplyAttachmentsCode(attachmentCode, true);
+
+			FirearmStatusFlags firearmStatusFlags = FirearmStatusFlags.MagazineInserted;
+			if (firearm.HasAdvantageFlag(AttachmentDescriptiveAdvantages.Flashlight))
+			{
+				firearmStatusFlags |= FirearmStatusFlags.FlashlightEnabled;
+			}
+			firearm.Status = new(firearm.AmmoManagerModule.MaxAmmo, firearmStatusFlags, firearm.GetCurrentAttachmentsCode());
+		}
+
+		/// <summary>
+		/// Applies a set of random attachments to the specified <see cref="Firearm"/>.
+		/// </summary>
+		/// <param name="firearm">The <see cref="Firearm"/> to which random attachments will be applied.</param>
+		/// <exception cref="ArgumentNullException">Thrown if the <paramref name="firearm"/> is null.</exception>
+		public static void ApllyRandomAttachments(this Firearm firearm)
+		{
+			if (firearm == null)
+				throw new ArgumentNullException(nameof(Firearm), "Firearm is null");
+
+			firearm.ApplyAttachmentsCode(AttachmentsUtils.GetRandomAttachmentsCode(firearm.ItemTypeId), true);
+
+			FirearmStatusFlags firearmStatusFlags = FirearmStatusFlags.MagazineInserted;
+			if (firearm.HasAdvantageFlag(AttachmentDescriptiveAdvantages.Flashlight))
+			{
+				firearmStatusFlags |= FirearmStatusFlags.FlashlightEnabled;
+			}
+			firearm.Status = new(firearm.AmmoManagerModule.MaxAmmo, firearmStatusFlags, firearm.GetCurrentAttachmentsCode());
+		}
+
+		/// <summary>
+		/// Applies the preferences of a <see cref="Player"/>'s attachments to a given <see cref="Firearm"/>.
+		/// </summary>
+		/// <param name="firearm">The <see cref="Firearm"/> to modify.</param>
+		/// <param name="player">The <see cref="Player"/> whose attachments preferences are to be applied.</param>
+		/// <param name="reloadWeapon">If this value is true, the weapon will be fully reloaded with ammunition when the attachment is applied.</param>
+		/// <returns>Returns true if the weapon was modified with the attachments; otherwise, false.</returns>
+		public static bool TryApplyAttachmentsPreferences(this Firearm firearm, Player player, bool reloadWeapon = false)
+		{
+			if(firearm == null)
+				throw new ArgumentNullException(nameof(Firearm), "Firearm is null");
+
+			if(AttachmentsServerHandler.PlayerPreferences.TryGetValue(player.ReferenceHub, out var prefs) && prefs.TryGetValue(firearm.ItemTypeId, out uint attachmentsCode))
+			{
+				firearm.ApplyAttachmentsCode(attachmentsCode, true);
+
+				FirearmStatusFlags firearmStatusFlags = FirearmStatusFlags.MagazineInserted;
+				if (firearm.HasAdvantageFlag(AttachmentDescriptiveAdvantages.Flashlight))
+				{
+					firearmStatusFlags |= FirearmStatusFlags.FlashlightEnabled;
+				}
+
+				if(reloadWeapon)
```

Comment:
Wouldnt this either force ammo to 0, or to the max, and not retain current ammunition?